### PR TITLE
Updated generation of Table of Contents to have a hierarchy and minor…

### PR DESCRIPTION
**Pull request by BeardedGnome**
… clean up.

Biggest change is adding parents and children to the schema references to help build the hierarchy for the Table of Contents.

@bghgary
@HowardWolosky

### CODE REVIEW

**Comment by emackey** [comment]
Thanks @BeardedGnome!  Before we can take code from a new contributor, we need you to fill out the [CLA (Contributor License Agreement)](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Documentation/Contributors/CLAs/individual-cla-agi-v1.0.txt) and email to cla@agi.com.  This just makes it explicit that the code was intended to be shared.  Thanks!

@HowardWolosky do you have bandwidth for a code review?

**Comment by HowardWolosky** [comment]
Yup.  I started the review already and will finish it in the next couple hours.

**Comment by pjcozzi** [comment]
Thanks all!

We now have a CLA from @BeardedGnome.

**Comment by emackey** [comment]
Thanks @BeardedGnome and @pjcozzi.

Back to you @HowardWolosky.  Now that the red tape is cut through, feel free to hit merge on this anytime you're happy with it.  Thanks!

**Comment by HowardWolosky** [comment]
I'm on it....will finish the review today.  Apologies for the delay.

File: lib/generateMarkdown.js [hunk 1]
```diff
@@ -42,11 +42,14 @@ function generateMarkdown(options) {
...
     var orderedTypes = sortObject(resolved.referencedSchemas);
+    for (var type in orderedTypes) {
+        orderedTypes[type].children.sort();
+    }
 
     // We need the reverse-sorted version so that when we do type searching we find the longest type first.
     var orderedTypesDescending = sortObject(resolved.referencedSchemas, false);
 
-    md += getTableOfContentsMarkdown(schema, orderedTypes, options.headerLevel);
+     md += getTableOfContentsMarkdown(schema, orderedTypes, options.headerLevel);
```
Comment (BeardedGnome) [inline lib/generateMarkdown.js]: I've gone through and used Shift-Alt-F to let VS Code auto format the documents.

File: lib/schema3Resolver.js [hunk 2]
```diff
@@ -31,8 +31,8 @@ function resolve(schema, fileName, basePath, ignorableTypes, debugOutputPath) {
         });
     }
 
-    referencedSchemas[schema.title] = { schema: schemaClone, fileName: fileName };
-    schemaClone = replaceRef(schemaClone, basePath, ignorableTypes, referencedSchemas);
+    referencedSchemas[schema.title] = { schema: schemaClone, fileName: fileName, parents: [], children: [] };
+    schemaClone = replaceRef(schemaClone, basePath, ignorableTypes, referencedSchemas, "root_object");
```
Comment (BeardedGnome) [inline lib/schema3Resolver.js]: Looks like it is unnecessary.

**Comment by HowardWolosky** [comment]
Thanks for your contribution, @BeardedGnome!  This definitely makes a more readable TOC for the documentation.

**Pull request merged**